File: tests/test_data/csharp_service/KafkaPatternTest.cs
using Confluent.Kafka;
using System.Threading.Tasks;
using Microsoft.Extensions.Configuration;

namespace TestService.Kafka
{
    [Topic("attribute-topic")]
    public class KafkaPatternTest
    {
        private readonly IProducer<string, string> producer;
        private readonly IConsumer<string, string> consumer;
        private readonly ProducerConfig producerConfig;
        private readonly ConsumerConfig consumerConfig;

        public KafkaPatternTest(IConfiguration config)
        {
            producerConfig = new ProducerConfig
            {
                Topic = "producer-config-topic",
                BootstrapServers = "localhost:9092"
            };

            consumerConfig = new ConsumerConfig
            {
                Topic = "consumer-config-topic",
                GroupId = "test-group",
                BootstrapServers = "localhost:9092"
            };

            producer = new ProducerBuilder<string, string>(producerConfig).Build();
            consumer = new ConsumerBuilder<string, string>(consumerConfig).Build();
        }

        // Producer patterns
        public async Task ProducerPatternsAsync()
        {
            // Direct producer methods
            await producer.ProduceAsync("direct-topic", new Message<string, string> { Value = "message" });
            producer.Produce("sync-topic", new Message<string, string> { Value = "message" });

            // With builder
            var builder = new ProducerBuilder<string, string>(producerConfig)
                .SetTopic("builder-topic")
                .WithTopic("with-topic");

            // Message patterns
            var message = new KafkaMessage("message-topic", "value");
            await messageProducer.SendAsync("send-topic", message);
        }

        // Consumer patterns
        [KafkaListener("listener-topic")]
        public void ListenerMethod(string message)
        {
        }

        public void ConsumerPatterns()
        {
            // Direct consumer methods
            consumer.Subscribe("direct-topic");
            consumer.Subscribe(new[] { "array-topic" });

            // With builder
            var builder = new ConsumerBuilder<string, string>(consumerConfig)
                .SetTopic("builder-topic")
                .WithTopic("with-topic");

            // Async consume
            await consumer.ConsumeAsync("async-topic");
        }
    }

    public class ProducerSettings 
    {
        public string Topic { get; set; } = "settings-topic";
    }

    public class ConsumerSettings
    {
        public string Topic { get; set; } = "settings-topic";
    }
}

File: tests/test_data/python_service/kafka_pattern_test.py
from kafka import KafkaProducer, KafkaConsumer
from confluent_kafka import Producer, Consumer
from typing import Dict
import json

# Constants and configuration
TOPIC = "constant-topic"
topic = "variable-topic"

producer_config = {
    "topic": "config-topic",
    "bootstrap.servers": "localhost:9092"
}

consumer_config = {
    "topic": "consumer-topic",
    "group.id": "test-group"
}

class KafkaPatternTest:
    def __init__(self):
        self.producer = KafkaProducer(bootstrap_servers="localhost:9092")
        self.consumer = KafkaConsumer(bootstrap_servers="localhost:9092")
        
        self.producer_config = ProducerConfig()
        self.producer_config.topic = "producer-config-topic"
        
        self.consumer_config = ConsumerConfig()
        self.consumer_config.topic = "consumer-config-topic"

    # Producer patterns
    def test_producers(self):
        # Direct producer methods
        self.producer.send("direct-topic", b"message")
        self.producer.produce("produce-topic", b"message")
        
        # Confluent producer
        conf_producer = Producer({"bootstrap.servers": "localhost:9092"})
        conf_producer.produce("confluent-topic", b"message")

    # Consumer patterns
    @kafka_listener(topic="decorator-topic")
    def listen_decorator(self, message):
        pass
        
    @consumer(topic="consumer-decorator-topic")
    def consumer_decorator(self, message):
        pass

    def test_consumers(self):
        # Direct consumer methods
        self.consumer.subscribe(["subscribe-topic"])
        self.consumer.subscribe({"pattern-topic.*"})
        
        # Confluent consumer
        conf_consumer = Consumer({"bootstrap.servers": "localhost:9092"})
        conf_consumer.subscribe(["confluent-topic"])

class ProducerConfig:
    def __init__(self):
        self.topic = "producer-init-topic"
        
class ConsumerConfig:
    def __init__(self):
        self.topic = "consumer-init-topic"
